### PR TITLE
Yoda watcher - JBWatcher, JBWatcherManager classes along with test classes were implemented

### PR DESCRIPTION
**Pull request by PeymanPejman**
RESOLVED - Test classes do not have package statement thus belong to the default unnamed package. I was not able conclusively determine why the package statement must be removed. It is not in line with conventions where test classes are part of the same package as the src classes.

### CODE REVIEW

File: jb-yoda/src/main/java/life/jukebox/JBWatcherManager.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import java.util.*;
+import java.util.concurrent.*;
+
+public class JBWatcherManager {
+  private ScheduledExecutorService scheduler;
+  private Set<JBWatcher> activeWatchers;
+  private static JBWatcherManager manager;
+  private final long WATCHER_PERIOD = 10;
```
Comment (peddybeats) [inline jb-yoda/src/main/java/life/jukebox/JBWatcherManager.java]: should be static

File: jb-yoda/src/main/java/life/jukebox/JBWatcherManager.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  public static JBWatcherManager getInstance() {
+    if (manager == null) {
+      manager = new JBWatcherManager();
+    }
+    return manager;
+  }
+
+  public void addWatcher(JBWatcher watcher) {
```
Comment (peddybeats) [inline jb-yoda/src/main/java/life/jukebox/JBWatcherManager.java]: Let's call this activateWatcher

File: jb-yoda/src/main/java/life/jukebox/JukeboxCreatedMessageReceiver.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package life.jukebox.yoda;
```
Comment (peddybeats) [inline jb-yoda/src/main/java/life/jukebox/JukeboxCreatedMessageReceiver.java:1]: Why is this a new file? It's there today..? Is your local repo's upstream updated?

File: jb-yoda/src/main/java/life/jukebox/YodaSubscriberClient.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
+package life.jukebox.yoda;
```
Comment (peddybeats) [inline jb-yoda/src/main/java/life/jukebox/YodaSubscriberClient.java:1]: Same with this - is your local repo's upstream updated?

File: jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  @Before 
+  public void createWather() {
+    watcher = new JBWatcher("test-JBId");
+    System.setOut(new PrintStream(outContent));
+  }
+
+  @Test
+  public void testRun() {
+    watcher.run();
+    assertNotNull(outContent.toString());
```
Comment (peddybeats) [inline jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java:23]: In general, be as specific with your tests as possible. Don't check for existence, check for validity of the message received.
Make the message watcher prints out a final public static string. Then here check that you're receiving that specific string.

File: jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    assertTrue(true);
+  }
+
+  @After
+  public void cleanUp(){
+    watcher = null;
```
Comment (peddybeats) [inline jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java]: You don't need to do this. watcher will automatically go out of scope with the rest of the resources as the test finishes.

File: jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.*;
+import java.io.*;
+
+public class JBWatcherTest {
+
+  private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+  JBWatcher watcher;
+  
+  @Before 
+  public void createWather() {
```
Comment (peddybeats) [inline jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java]: rename this to setup - you're doing more than createWatcher

File: jb-yoda/src/main/java/life/jukebox/JBWatcher.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import java.util.concurrent.*;
+
+public class JBWatcher implements Runnable{
+  private ScheduledFuture<?> watcherHandle;
```
Comment (peddybeats) [inline jb-yoda/src/main/java/life/jukebox/JBWatcher.java]: watcherHandle doesn't communicate this is the handle to the scheduled task that runs this watcher - maybe taskHandle?

File: jb-yoda/src/main/java/life/jukebox/JBWatcher.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+  public ScheduledFuture<?> getWatcherHandle(){
+    return watcherHandle;
+  } 
+
+  public void setWatcherHandle(ScheduledFuture<?> watcherHandle){
+    this.watcherHandle = watcherHandle;
+  }
+
+  public void run(){
```
Comment (peddybeats) [inline jb-yoda/src/main/java/life/jukebox/JBWatcher.java]: Javadoc on all non-trivial (basic getters and setters) public methods. Please do this for all.

**Review by peddybeats — CHANGES_REQUESTED** [review]

File: jb-yoda/src/main/java/life/jukebox/JBWatcher.java (hunk 8 above)
Comment (PeymanPejman) [reply jb-yoda/src/main/java/life/jukebox/JBWatcher.java]: fixed

File: jb-yoda/src/main/java/life/jukebox/JBWatcherManager.java (hunk 1 above)
Comment (PeymanPejman) [reply jb-yoda/src/main/java/life/jukebox/JBWatcherManager.java]: fixed

File: jb-yoda/src/main/java/life/jukebox/JBWatcherManager.java (hunk 2 above)
Comment (PeymanPejman) [reply jb-yoda/src/main/java/life/jukebox/JBWatcherManager.java]: fixed

File: jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java (hunk 7 above)
Comment (PeymanPejman) [reply jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java]: fixed

File: jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java (hunk 6 above)
Comment (PeymanPejman) [reply jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java]: fixed

File: jb-yoda/src/main/java/life/jukebox/JBWatcher.java (hunk 9 above)
Comment (PeymanPejman) [reply jb-yoda/src/main/java/life/jukebox/JBWatcher.java]: fixed

File: jb-yoda/src/main/java/life/jukebox/JukeboxCreatedMessageReceiver.java (hunk 3 above)
Comment (PeymanPejman) [reply jb-yoda/src/main/java/life/jukebox/JukeboxCreatedMessageReceiver.java:1]: updated upstream. Not sure why it happened.

File: jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java (hunk 5 above)
Comment (PeymanPejman) [reply jb-yoda/src/test/java/life/jukebox/JBWatcherTest.java:23]: Addressed in private msg. Will not be fixing.

**Review by peddybeats — APPROVED** [review]
LGTM

**Pull request merged**